pkg/asset/installconfig/installconfig_test.go

```diff
@@ -0,0 +1,188 @@
+package installconfig
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+type testAsset struct {
+	name string
+}
+
+func (a *testAsset) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+func (a *testAsset) Generate(map[asset.Asset]*asset.State) (*asset.State, error) {
+	return nil, nil
+}
+
+func TestInstallConfigDependencies(t *testing.T) {
+	stock := &StockImpl{
+		clusterID:    &testAsset{name: "test-cluster-id"},
+		emailAddress: &testAsset{name: "test-email"},
+		password:     &testAsset{name: "test-password"},
+		baseDomain:   &testAsset{name: "test-domain"},
+		clusterName:  &testAsset{name: "test-cluster"},
+		license:      &testAsset{name: "test-license"},
+		pullSecret:   &testAsset{name: "test-pull-secret"},
+		platform:     &testAsset{name: "test-platform"},
+	}
+	installConfig := &installConfig{
+		assetStock: stock,
+	}
+	exp := []string{
+		"test-cluster-id",
+		"test-email",
+		"test-password",
+		"test-domain",
+		"test-cluster",
+		"test-license",
+		"test-pull-secret",
+		"test-platform",
+	}
+	deps := installConfig.Dependencies()
+	act := make([]string, len(deps))
+	for i, d := range deps {
+		a, ok := d.(*testAsset)
+		assert.True(t, ok, "expected dependency to be a *testAsset")
+		act[i] = a.name
+	}
+	assert.Equal(t, exp, act, "unexpected dependency")
+}
+
+func TestInstallConfigGenerate(t *testing.T) {
+	cases := []struct {
+		name                 string
+		platformContents     []string
+		expectedPlatformYaml string
+	}{
+		{
+			name: "aws",
+			platformContents: []string{
+				"aws",
+				"test-region",
+				"test-keypairname",
+			},
+			expectedPlatformYaml: `  aws:
+    keyPairName: test-keypairname
+    region: test-region
+    vpcCIDRBlock: ""
+    vpcID: ""`,
+		},
+		{
+			name: "libvirt",
+			platformContents: []string{
+				"libvirt",
+				"test-uri",
+				"test-sshkey",
+			},
+			expectedPlatformYaml: `  libvirt:
+    URI: test-uri
+    masterIPs: null
+    network:
+      if: ""
+      ipRange: ""
+      name: ""
+      resolver: ""
+    sshKey: test-sshkey`,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			stock := &StockImpl{
+				clusterID:    &testAsset{},
+				emailAddress: &testAsset{},
+				password:     &testAsset{},
+				baseDomain:   &testAsset{},
+				clusterName:  &testAsset{},
+				license:      &testAsset{},
+				pullSecret:   &testAsset{},
+				platform:     &testAsset{},
+			}
+
+			dir, err := ioutil.TempDir("", "TestInstallConfigGenerate")
+			if err != nil {
+				t.Skipf("could not create temporary directory: %v", err)
+			}
+			defer os.RemoveAll(dir)
+
+			installConfig := &installConfig{
+				assetStock: stock,
+				directory:  dir,
+			}
+
+			states := map[asset.Asset]*asset.State{
+				stock.clusterID: &asset.State{
```

Comment:
We currently don't run `gofmt` in CI (I've got work in flight to add that, #173), but running it locally complains about this line and the later ones following the same pattern:

```console
$ git describe --always --abbrev=0
1273983432bf1ee9dbf51cd71cf8aaf7a6be798b
$ find . -name '*.go' ! -path '*/vendor/*' ! -path '*/.build/*' -exec gofmt -s -w {} \+
$ git diff --exit-code pkg
diff --git a/pkg/asset/installconfig/installconfig_test.go b/pkg/asset/installconfig/installconfig_test.go
index 32fcb58..0355717 100644
--- a/pkg/asset/installconfig/installconfig_test.go
+++ b/pkg/asset/installconfig/installconfig_test.go
@@ -120,28 +120,28 @@ func TestInstallConfigGenerate(t *testing.T) {
 			}
 
 			states := map[asset.Asset]*asset.State{
-				stock.clusterID: &asset.State{
+				stock.clusterID: {
 					Contents: []asset.Content{{Data: []byte("test-cluster-id")}},
 				},
-				stock.emailAddress: &asset.State{
+				stock.emailAddress: {
 					Contents: []asset.Content{{Data: []byte("test-email")}},
 				},
...
```
```

Reply:
Fixed via #181.